Data generator reorganisation

shounakmk219 opened the pull request:
## Description
This PR decouples the data writer part from the data generator so that we can easily add new writers by keeping data generator untouched.

## Labels
`refactor`

xiangfu0 commented on pinot-controller/src/main/java/org/apache/pinot/controller/recommender/data/writer/AvroRecordAppender.java:
```diff
@@ -0,0 +1,56 @@
+  private final DataFileWriter<GenericData.Record> _recordWriter;
+  private final org.apache.avro.Schema _avroSchema;
+
+  public AvroRecordAppender(File file, org.apache.avro.Schema avroSchema)
```
Is this class reusable? Or shall we merge it into AvroWriter?

shounakmk219 replied:
Atleast right now only AvroWriter is using it so makes sense to merge👍

xiangfu0 commented on pinot-controller/src/main/java/org/apache/pinot/controller/recommender/data/writer/FileWriter.java:
```diff
@@ -0,0 +1,53 @@
+    }
+  }
+
+  protected String generateRow(DataGenerator generator) {
```
Make this method/class abstract?

shounakmk219 replied:
Agree, will make the change.

xiangfu0 reviewed (approved):
overall lgtm, can you please add more comments to the interfaces you added?

shounakmk219 commented:
Thanks for the review @xiangfu0 !
I have addressed your comments along with a bugfix for removing hardcoded file extension and also added a new `cleanup()` method to `Writer` interface.
